cocos/scripting/js-bindings/manual/jsb_cocos2dx_manual.cpp

```diff
@@ -836,6 +819,89 @@ static bool js_EventListenerCustom_create(se::State& s)
 }
 SE_BIND_FUNC(js_EventListenerCustom_create)
 
+static void onBeforeDispatchTouchEvent(Event* event)
+{
+    if (event->getType() == Event::Type::TOUCH)
+    {
+        EventTouch* touchEvent = static_cast<EventTouch*>(event);
+        if (touchEvent->getEventCode() == EventTouch::EventCode::BEGAN)
+        {
+            bool fromCache = true;
+            const auto& touches = touchEvent->getTouches();
+            for (auto&& touch : touches)
+            {
+                se::Value touchVal;
+                native_ptr_to_seval<Touch>(touch, __jsb_cocos2d_Touch_class, &touchVal, &fromCache);
+                assert(!fromCache);
+                touchVal.toObject()->root();
+            }
+        }
+    }
+}
+
+static void onAfterDispatchTouchEvent(Event* event)
+{
+    if (event->getType() == Event::Type::TOUCH)
+    {
+        EventTouch* touchEvent = static_cast<EventTouch*>(event);
+        EventTouch::EventCode code = touchEvent->getEventCode();
+        if (code == EventTouch::EventCode::ENDED || code == EventTouch::EventCode::CANCELLED)
+        {
+            const auto& touches = touchEvent->getTouches();
+            for (auto&& touch : touches)
+            {
+                auto iter = se::NativePtrToObjectMap::find(touch);
+                assert(iter != se::NativePtrToObjectMap::end());
+                iter->second->unroot();
+            }
+        }
+    }
+}
+
+static void onTextureCreate(TextureCache* cache, Texture2D* texture)
+{
+    auto iterOwner = se::NativePtrToObjectMap::find(cache);
+    if (iterOwner == se::NativePtrToObjectMap::end())
+    {
+        assert(false);
+        return;
+    }
+
+    se::Value textureVal;
+    bool fromCache = false;
+    native_ptr_to_seval<Texture2D>(texture, __jsb_cocos2d_Texture2D_class, &textureVal, &fromCache);
+    if (!fromCache && textureVal.isObject())
+    {
+        iterOwner->second->attachObject(textureVal.toObject());
```

Comment:
After a texture being added to texturecache. Create a js object to associate with it to avoid dangerous se::Object being used.